### DISCUSSION TITLE
Anyone using MathML with Markdoc?

### DISCUSSION THREAD

**Discussion by vanakenm** [markdoc/markdoc#316] [Q&A] [answered]
Hi,
[MathML](https://en.wikipedia.org/wiki/MathML) allows for proper rendering of mathematical formula. Some implementation of Markdown does have a mathml impl - for example VSCode:

![image](https://user-images.githubusercontent.com/829292/214343320-1b184d36-45e1-4a83-a2a7-1566dd22a04f.png) - https://github.com/microsoft/vscode-docs/blob/vnext/release-notes/v1_58.md#math-formula-rendering-in-the-markdown-preview

Did anyone try to get some MathML in Markdoc (it does not seems to work OOTB) - any idea how to get this to work?

Thanks!

**Comment by mfix-stripe** [comment, marked as answer]
@vanakenm I would suggest [creating a `math` tag](https://markdoc.dev/docs/tags):
```md
{% math %}

\displaystyle
\left( \sum_{k=1}^n a_k b_k \right)^2
\leq
\left( \sum_{k=1}^n a_k^2 \right)
\left( \sum_{k=1}^n b_k^2 \right)

{% /math %}
```

**Reply by vanakenm** [reply]
Did just that, using react-katex to do the actual rendering. Works great!
Don't know if markdoc has some kind of "plugin" system, but that would be nice to have there.

Thanks!

**Reply by mfix-stripe** [reply]
@vanakenm you can think of custom tags, nodes, functions, etc. as "plugins" that can be plugged-in on a case by case basis. Markdoc has many patterns for extensibility, though, even the option to create your own custom renderer. 
More info over in the docs: https://markdoc.dev/docs
